src/subscript/welltest_extract/welltest_extract.py

```diff
@@ -0,0 +1,396 @@
+
+    dp_lag2_forward = np.hstack((dp_lag2, dp[-1], 0))  # As for dspt_forward
+
+    dspt_lag2_backward = np.hstack((1, dspt[0], dspt_lag2))
+    # First backward step does not exist. Second backward step is only one step
+
+    dp_lag2_backward = np.hstack((0, dp[0], dp_lag2))  # As for dspt_backward
+
+    # Calculate the lag2 weighted derivative
+    dpdspt_lag2_forward = dp_lag2_forward / dspt_lag2_forward
+    dpdspt_lag2_backward = dp_lag2_backward / dspt_lag2_backward
+
+    # To get right values in denominator. It has value 1 above,
+    # to avoid division on zero.
+    dspt_lag2_forward[0] = 0
+    dspt_lag2_backward[-1] = 0
+
+    dpdspt_weighted_lag2 = (
+        dpdspt_lag2_forward * dspt_lag2_backward
+        + dpdspt_lag2_backward * dspt_lag2_forward
+    ) / (dspt_lag2_backward + dspt_lag2_forward)
+
+    return dpdspt_weighted_lag2
+
+
+def to_csv(filen, field_list, header_list, sep=","):
+    """
+    Dump vectors to csv file. Handles arbitrarly number of fields
+    """
+
+    fileh = open(filen, "w")
+
+    fileh.write(header_list[0])
+    for header in header_list[1:]:
+        fileh.write(sep + header)
+    fileh.write("\n")
+    for i in range(len(field_list[0])):
+        fileh.write("%0.10f" % field_list[0][i])
+        for field in field_list[1:]:
+            fileh.write(sep + "%0.10f" % field[i])
+        fileh.write("\n")
+    fileh.close()
+
+
+def main():
+    """
+    Main entry point for the script
+    """
+
+    print("Running the " + sys.argv[0])
+    args = get_parser().parse_args()
+
+    eclcase = args.eclcase
+    well_name = args.well_name
+    buildup_nr = args.buildup_nr
+    main_phase = args.phase
+    outf_prefix = args.out_files_prefix
+
+    summary = EclSum(eclcase)
+
+    wbhp = get_summary_vec(summary, "WBHP:" + well_name)
+    wwpr = get_summary_vec(summary, "WWPR:" + well_name)
+    wopr = None
+    wgpr = None
+    if main_phase == "OIL":
+        wopr = get_summary_vec(summary, "WOPR:" + well_name)
+    else:
+        wgpr = get_summary_vec(summary, "WGPR:" + well_name)
+    time = np.array(summary.days) * 24.0
+
+    if main_phase == "OIL":
+        buildup_indices, buildup_end_indices = get_buildup_indices(wopr)
+    else:
+        buildup_indices, buildup_end_indices = get_buildup_indices(wgpr)
+
+    print("Time step number for start of each build-up period " + str(buildup_indices))
+    print(
+        "Time step number for end   of each build-up period "
+        + str(buildup_end_indices)
+        + "\n"
+    )
+
+    if buildup_nr > len(buildup_indices):
+        sys.stderr.write(
+            "There are %d build ups detected for well %s. You asked for nr %d\n"
+            % (len(buildup_indices), well_name, buildup_nr)
+        )
+        sys.exit(1)
+    else:
+        bu_start_ind = buildup_indices[buildup_nr - 1]
+        print(
+            "The time step for the start of the %d'th build-up period is %d"
```

Comment:
"build up", "buildup" or "build-up". We should decide.. Suggest "buildup"

Reply:
Fixed!

With regards the multiple csv files; this has been nagging me as well, but I have not been able to find a good solution. The files with one column are typically used as input to ERTs GEN_DATA, and are not csv files as such. The files with two columns, are used for visualization in eg Spotfire I guess. These could probably be merged into one unified file. We are still left with at least 4 files, and I don't like it. Another issue is wwpr, which is parsed and exported to file, but never used by the script. I would have preferred not having it as a requirement. An option would be to complexify the cli options and let the user configure.